src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -83,6 +83,14 @@ private SortOption SortOption
         /// </summary>
         private readonly Dictionary<IVsTextView, ITextView> _trackedTextViews = new();
 
+        /// <summary>
+        /// Returns whether SymbolTree.ItemsSource has been initialized. 
+        /// </summary>
+        /// <remarks>
+        /// It is only safe to read/mutate SymbolTreeIsInitialized from the UI thread.
```

Comment:
👍

Reply:
note: you could consider enforcing this by having this be a full get/set and having both check that that is true.  up to you.

Reply:
+1, that's part of the benefit of properties is that we can add checks like this